File: Symfony/src/Entity/Event.php
<?php

namespace App\Entity;

use DateTime;
use Doctrine\DBAL\Types\Types;
use App\Repository\EventRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;

use Doctrine\Common\Collections\Collection;
use Doctrine\Common\Collections\ArrayCollection;


#[ORM\Entity(repositoryClass: EventRepository::class)]
class Event
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "AUTO")]
    #[ORM\Column(type: "integer")]
    #[Groups(['event:read'])]
    private ?int $idevent;

    #[ORM\ManyToOne(targetEntity:"Etablissement")]
    #[ORM\JoinColumn(name:"idEstab", referencedColumnName:"ID_Etablissement")]
    #[Assert\NotBlank(message: "L'ID de l'établissement ne doit pas être vide.")]
    #[Groups(['event:read'])]
    private $idestab;

    #[ORM\Column(length :255)]
    #[Assert\NotBlank(message: "Le nom de l'événement ne doit pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "Le nom de l'événement ne peut pas dépasser {{ limit }} caractères.")]
    #[Groups(['event:read'])]
    private ?string $nameevent;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotBlank(message: "La date de l'événement ne doit pas être vide.")]
    #[Assert\GreaterThan("today", message: "La date de l'événement doit être ultérieure à la date actuelle.")]
    #[Assert\Type(type: "\DateTimeInterface", message: "La date de l'événement doit être de type date.")]
    #[Groups(['event:read'])]
    private ?\DateTimeInterface $dateevent;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Le nombre maximum ne doit pas être vide.")]
    #[Assert\GreaterThan(value: 0, message: "Le nombre maximum doit être supérieur à zéro.")]
    #[Groups(['event:read'])]
    private ?int $nbrmax;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Le prix ne doit pas être vide.")]
    #[Assert\GreaterThan(value: 0, message: "Le prix doit être supérieur à zéro.")]
    #[Groups(['event:read'])]
    private ?float $prix;

    #[ORM\Column(length :500)]
    #[Assert\NotBlank(message: "La description ne doit pas être vide.")]
    #[Assert\Length(max: 500, maxMessage: "La description ne peut pas dépasser {{ limit }} caractères.")]
    #[Groups(['event:read'])]
    private ?string $description;

    #[ORM\Column(length :255)]
    //#[Assert\NotBlank(message: "Le chemin de l'image ne doit pas être vide.")]
    //#[Assert\Length(max: 255, maxMessage: "Le chemin de l'image ne peut pas dépasser {{ limit }} caractères.")]
    #[Groups(['event:read'])]
    private ?string $image;

    
    #[ORM\ManyToOne(targetEntity:"Partner")]
    #[ORM\JoinColumn(name:"idPartnerCE", referencedColumnName:"idPartner")]
    #[Groups(['event:read'])]
    private ?Partner $idpartnerce;

    public function getIdevent(): ?int
    {
        return $this->idevent;
    }

       /**
     * @return mixed
     */
    public function getIdestab()
    {
        return $this->idestab;
    }

    /**
     * @param mixed $idestab
     */
    public function setIdestab($idestab): void
    {
        $this->idestab = $idestab;
    }


    public function getNameevent(): ?string
    {
        return $this->nameevent;
    }

    public function setNameevent(string $nameevent): static
    {
        $this->nameevent = $nameevent;

        return $this;
    }

    public function getDateevent(): ?\DateTimeInterface
    {
        return $this->dateevent;
    }

    public function setDateevent(\DateTimeInterface $dateevent): static
    {
        $this->dateevent = $dateevent;

        return $this;
    }

    public function getNbrmax(): ?int
    {
        return $this->nbrmax;
    }

    public function setNbrmax(int $nbrmax): static
    {
        $this->nbrmax = $nbrmax;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): static
    {
        $this->image = $image;

        return $this;
    }

    public function getIdpartnerce(): ?Partner
    {
        return $this->idpartnerce;
    }

    public function setIdpartnerce(?Partner $idpartnerce): static
    {
        $this->idpartnerce = $idpartnerce;

        return $this;
    }


// **************************************************************************************
    #[ORM\OneToMany(targetEntity:"Reservation", mappedBy:"idEvent")]
    private $reservations;

    public function __construct()
    {
        $this->reservations = new ArrayCollection();
    }

   
    public function getReservations(): Collection
    {
        return $this->reservations;
    }
    
    public function getTotalReservedPlaces(): int
{
    $reservations = $this->getReservations(); // Assuming you have a method to get reservations associated with the event
    $totalReservedPlaces = 0;

    foreach ($reservations as $reservation) {
        $totalReservedPlaces += $reservation->getNbPlaces();
    }

    return $totalReservedPlaces;
}

public function isFullyBooked(): bool
{
    return $this->getTotalReservedPlaces() >= $this->getNbrmax();
}

}

File: JAVA/src/main/java/models/Event.java
package models;

import java.sql.Date;
import java.util.Objects;

public class Event {

    private int idEvent;
    private int idPartner;
    private int idEstab;
    private String nameEvent;

    private Date dateEvent;
    private int nbrMax;
    private double prix;
    private String description;

    private String image;

    public Event() {
    }

    public Event(int idEvent,int idPartner, int idEstab, String nameEvent, Date dateEvent, int nbrMax,double prix, String description , String image) {
        this.idEvent = idEvent;
        this.idPartner = idPartner;
        this.idEstab = idEstab;
        this.nameEvent = nameEvent;
        this.dateEvent = dateEvent;
        this.nbrMax = nbrMax;
        this.prix = prix;
        this.description = description;
        this.image = image;
    }

    public Event(int idPartner, int idEstab, String nameEvent, Date dateEvent, int nbrMax,double prix, String description , String image) {
        this.idPartner = idPartner;
        this.idEstab = idEstab;
        this.nameEvent = nameEvent;
        this.dateEvent = dateEvent;
        this.nbrMax = nbrMax;
        this.prix = prix;
        this.description = description;
        this.image = image;
    }


    public Event(int idEstab, String nameEvent, Date dateEvent, int nbrMax,double prix, String description, String image) {
        this.idEstab = idEstab;
        this.nameEvent = nameEvent;
        this.dateEvent = dateEvent;
        this.nbrMax = nbrMax;
        this.prix = prix;
        this.description = description;
        this.image = image;

    }

    public int getIdEvent() {
        return idEvent;
    }

    public void setIdEvent(int idEvent) {
        this.idEvent = idEvent;
    }

    public int getIdPartner() {
        return idPartner;
    }

    public void setIdPartner(int idPartner) {
        this.idPartner = idPartner;
    }

    public int getIdEstab() {
        return idEstab;
    }

    public void setIdEstab(int idEstab) {
        this.idEstab = idEstab;
    }

    public String getNameEvent() {
        return nameEvent;
    }

    public void setNameEvent(String nameEvent) {
        this.nameEvent = nameEvent;
    }

    public Date getDateEvent() {
        return dateEvent;
    }

    public void setDateEvent(Date dateEvent) {
        this.dateEvent = dateEvent;
    }

    public int getNbrMax() {
        return nbrMax;
    }

    public void setNbrMax(int nbrMax) {
        this.nbrMax = nbrMax;
    }


    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }
    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    @Override
    public String toString() {
        return "Event{" +
                "idEvent=" + idEvent +
                ", idPartner=" + idPartner +
                ", idEstab=" + idEstab +
                ", nameEvent='" + nameEvent + '\'' +
                ", dateEvent=" + dateEvent +
                ", nbrMax=" + nbrMax +
                ", prix=" + prix +
                ", description='" + description + '\'' +
                ", image='" + image + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Event event)) return false;
        return getIdEvent() == event.getIdEvent() && getIdPartner() == event.getIdPartner() && getIdEstab() == event.getIdEstab() && getNbrMax() == event.getNbrMax() && Double.compare(prix, event.prix) == 0 && Objects.equals(getNameEvent(), event.getNameEvent()) && Objects.equals(getDateEvent(), event.getDateEvent()) && Objects.equals(getDescription(), event.getDescription()) && Objects.equals(getImage(), event.getImage());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getIdEvent(), getIdPartner(), getIdEstab(), getNameEvent(), getDateEvent(), getNbrMax(), prix, getDescription(), getImage());
    }
}
